kedro/extras/datasets/snowflake/snowflake_dataset.py

```diff
@@ -0,0 +1,252 @@
+                To find all supported connection string formats, see here:
+                https://docs.sqlalchemy.org/en/13/core/engines.html#database-urls
+            save_args: Provided to underlying pandas ``to_sql`` function along
+                with the connection string.
+                To find all supported arguments, see here:
+                https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_sql.html
+                To find all supported connection string formats, see here:
+                https://docs.sqlalchemy.org/en/13/core/engines.html#database-urls
+                It has ``index=False`` in the default parameters.
+
+        Raises:
+            DataSetError: When either ``table_name`` or ``con`` is empty.
+        """
+
+        if not table_name:
+            raise DataSetError("'table_name' argument cannot be empty.")
+
+        if not credentials:
+            raise DataSetError("Please configure expected credentials")
+
+        # print(self._load_args)
+
+        # Handle default load and save arguments
+        self._load_args = copy.deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        self._load_args["table_name"] = table_name
+        self._save_args["name"] = table_name
+
+        self._credentials = credentials["credentials"]
+
+        # self._connection_str = credentials["con"]
+        self._session = self._get_session(self._credentials)
+
+    @classmethod
+    def _get_session(cls, credentials: dict) -> None:
+        """Given a connection string, create singleton connection
+        to be used across all instances of `SQLQueryDataSet` that
+        need to connect to the same source.
+            connection_params = {
+                "account": "",
+                "user": "",
+                "password": "",
+                "role": "",
+                "warehouse": "",
+                "database": "",
+                "schema": ""
+                }
+        """
+        try:
+            session = Session.builder.configs(credentials).create()
+        except ImportError as import_error:
+            raise _get_missing_module_error(import_error) from import_error
+        except Exception as exception:
+            raise exception
+        return session
+
+    def _describe(self) -> Dict[str, Any]:
+        load_args = copy.deepcopy(self._load_args)
+        save_args = copy.deepcopy(self._save_args)
+        return dict(
+            table_name=self._load_args["table_name"],
+            load_args=load_args,
+            save_args=save_args,
+        )
+
+    def _load(self) -> pd.DataFrame:
+        sp_df = self._session.table(self._load_args["table_name"])
+        return sp_df.to_pandas()
+
+    def _save(self, data: pd.DataFrame) -> None:
```

Comment:
```suggestion
    def _save(self, data: [pd.DataFrame, snowpark.DataFrame]) -> None:
```

I'm not actually sure on the type signature, but the push here is to accept either option and handle gracefully.